packages/phosphor/src/TabPanel.ts

```diff
@@ -0,0 +1,87 @@
+import { HTMLWidget, SVGWidget, Widget } from "@hpcc-js/common";
+import { IMessageHandler, Message, TabPanel as PTabPanel, Widget as PWidget } from "@hpcc-js/phosphor-shim";
+import { Msg, WidgetAdapter } from "./WidgetAdapter";
+
+import "../src/DockPanel.css";
+
+export class TabPanel extends HTMLWidget {
+    private _tab = new PTabPanel({ tabPlacement: "top" });
+    protected content: WidgetAdapter[] = [];
+
+    constructor() {
+        super();
+        this._tag = "div";
+        this._tab.id = "p" + this.id();
+    }
+
+    protected getWidgetAdapter(widget: Widget): WidgetAdapter | null {
+        let retVal = null;
+        this.content.some(wa => {
+            if (wa.widget === widget) {
+                retVal = wa;
+                return true;
+            }
+            return false;
+        });
+        return retVal;
+    }
+
+    addWidget(widget: SVGWidget | HTMLWidget, title: string) {
```

Comment:
Shouldn't this be `widget: Widget`?

Reply:
Not sure why I went with that choice, I didn't test the CanvasWidget...  But will change in the future if needed.